### PR TITLE
Improve serialization and its unit tests

### PR DESCRIPTION
**Pull request by surgura**

### CODE REVIEW

File: src/Genes.h [hunk 1]
```diff
@@ -651,12 +672,44 @@ namespace NEAT
             return (a_lhs.m_InnovationID != a_rhs.m_InnovationID);
         }
 
-        friend bool operator==(const LinkGene &a_lhs, const LinkGene &a_rhs)
+        // friend bool operator==(const LinkGene &a_lhs, const LinkGene &a_rhs)
```
Comment (portaloffreedom) [inline src/Genes.h]: Remove this commented code?

File: tests/serialization/serialize_genome.cpp [hunk 2]
```diff
@@ -9,55 +9,34 @@
...
-            2, 1, 3,
-            NEAT::ActivationFunction::SIGNED_SIGMOID,
-            NEAT::ActivationFunction::SIGNED_SIGMOID,
-            params
-            );
-
-    const std::string serialized = serialize(g);
-    NEAT::Genome copy = deserialize(serialized);
-
-    BOOST_TEST(g.GetID() == copy.GetID());
```
Comment (portaloffreedom) [inline tests/serialization/serialize_genome.cpp:49]: I would keep these tests as well, just so we test if we ever break the `operator==` for Genome. What do you think?

**Review by portaloffreedom — CHANGES_REQUESTED** [review]
small changes maybe?

File: src/Genes.h (hunk 1 above)
Comment (surgura) [reply src/Genes.h]: Fixed

File: tests/serialization/serialize_genome.cpp (hunk 2 above)
Comment (surgura) [reply tests/serialization/serialize_genome.cpp:49]: Part of old tests restored

**Review by portaloffreedom — APPROVED** [review]

**Pull request merged**